### DISCUSSION TITLE
Issue with Running Scripts via API on Tactical RMM: "Method GET Not Allowed" Error

### DISCUSSION THREAD

**Discussion by DeGrootePasja** [amidaware/tacticalrmm#1986] [Q&A] [answered]
Hi everyone,

I've recently started using Tactical RMM and have encountered a problem when trying to run scripts through the API. I'm attempting to use the endpoint:

`https://api-url/agents/<agent_id>/runscript`

I include the "x-api-key" in the header and a JSON body that specifies the script, following the guidelines provided in the [Tactical RMM API documentation](https://docs.tacticalrmm.com/functions/api/#running-a-script-on-an-agent-using-the-api).

Despite this, I'm unable to execute the script as expected. Here's the basic format of my request:

Headers:
`
x-api-key: [Your-API-Key]
`
Body:
`
{
  "script_id": [script-id],
  "args": [arguments],
  "timeout": [timeout]
}
`
Error Message:
`{
    "detail": "Method \"GET\" not allowed."
}`
This response suggests there might be a method mismatch in my request. I am using "POST" in the request but I still get the error about "GET" and I am following the guidelines provided in the documentation. If anyone has experienced a similar issue or can spot what I might be doing wrong, your insights would be incredibly helpful.

Thanks in advance for your help!

**Comment by wh1te909** [comment]
paste your script here so we can see

**Comment by DeGrootePasja** [comment]
This script has been instrumental in my recent projects. Please note that placeholders marked with '<>' were replaced with actual data, not mock data. Below, you'll find the output generated by this script.

![image](https://github.com/user-attachments/assets/0d3ff197-22eb-4761-87d0-2beea0113aa4)

**Reply by wh1te909** [reply, marked as answer]
missing a trailing slash at the end of the url

**Reply by DeGrootePasja** [reply]
Thank you very much!

**Comment by Alphanumerisch** [comment]
I would like to join you.
I get a server error.

https://api.domain.de/agents/NDg57muPOoE3c4hzclUDwocgdnpAvyIWrlMlZjM/runscript/

The Body: 
{ 
   "script": "129"
}

GET requests work perfectly


Error Message:


<!doctype html>
<html lang="en">

<head>
	<title>Server Error (500)</title>
</head>

<body>
	<h1>Server Error (500)</h1>
	<p></p>
</body>

</html>


There is nothing relevant in the trmm_debug.log

**Reply by wh1te909** [reply]
you're looking at the wrong log file: https://docs.tacticalrmm.com/troubleshooting/#error-500

**Comment by Alphanumerisch** [comment]
thanks for answer.

I have modified the Body

{
  "env_vars": [],  
  "run_as_user": "$false", 
  "output": "wait",
  "script": "129",
  "timeout": "20",
  "args": [],
  "save_all_output": "$false"
}

I get no errors but a timeout.

**Comment by Alphanumerisch** [comment]
TRMM Version 0.20.0

I have tested it myself in trmm as Web Hok. The domain is resolved internally. we have a split dns setup.

The test returns the following message

URL: HTTPSConnectionPool(host='api.domain.de', port=443): Read timed out. (read timeout=8)

Body
HTTPSConnectionPool(host='api.domain.de', port=443): Read timed out. (read timeout=8)

Response
HTTPSConnectionPool(host='api.domain.de', port=443): Read timed out. (read timeout=8)
